### PR TITLE
fix: pause prompt stdin

### PR DESCRIPTION
**Pull request by RasPhilCo**

### CODE REVIEW

File: src/prompt.ts [hunk 1]
```diff
@@ -81,6 +81,9 @@ function normal(options: IPromptConfig, retries = 100): Promise<string> {
         resolve(data || options.default)
       }
     })
-    setTimeout(() => reject(), options.timeout || 10000).unref()
+    setTimeout(() => {
+      process.stdin.end()
+      reject(new Error('Prompt timeout'))
```
Comment (RasPhilCo) [inline src/prompt.ts]: not having an error object in reject was causing issues later in oclif/config

File: src/prompt.ts [hunk 2]
```diff
@@ -81,6 +81,9 @@ function normal(options: IPromptConfig, retries = 100): Promise<string> {
         resolve(data || options.default)
       }
     })
-    setTimeout(() => reject(), options.timeout || 10000).unref()
+    setTimeout(() => {
+      process.stdin.end()
```
Comment (RasPhilCo) [inline src/prompt.ts]: missing piece. 🍕

File: src/prompt.ts (hunk 1 above)
Comment (jdxcode) [reply src/prompt.ts]: I feel that's something TypeScript should enforce (possibly behind a flag)

File: src/prompt.ts (hunk 2 above)
Comment (jdxcode) [reply src/prompt.ts]: this would prevent us from reading stdin ever again in this process which seems problematic

Comment (jdxcode) [reply src/prompt.ts]: would `process.stdin.pause()` work?

File: src/prompt.ts [hunk 3]
```diff
@@ -81,8 +81,9 @@ function normal(options: IPromptConfig, retries = 100): Promise<string> {
         resolve(data || options.default)
       }
     })
-    if (options.timeout) {
```
Comment (jdxcode) [inline src/prompt.ts]: is there a reason you removed this conditional?

Comment (jdxcode) [reply src/prompt.ts]: I put it in to fix this issue: https://github.com/heroku/cli/issues/880

Comment (RasPhilCo) [reply src/prompt.ts]: merge mistake, putting it back

File: src/prompt.ts [hunk 4]
```diff
@@ -82,7 +82,10 @@ function normal(options: IPromptConfig, retries = 100): Promise<string> {
       }
     })
     if (options.timeout) {
-      setTimeout(() => reject(new Error('timed out')), options.timeout).unref()
+      setTimeout(() => {
+        process.stdin.pause()
+        reject(new Error('Prompt timeout'))
+      }, options.timeout || 10000).unref()
```
[the remark this answers is not in the document]
Comment (RasPhilCo) [reply src/prompt.ts]: lol

**Pull request merged**